src/Disks/ObjectStorages/S3/registerDiskS3.cpp

```diff
@@ -91,11 +118,19 @@ void registerDiskS3(DiskFactory & factory)
         FileCachePtr cache = getCachePtrForDisk(name, config, config_prefix, context);
         S3Capabilities s3_capabilities = getCapabilitiesFromConfig(config, config_prefix);
 
-        ObjectStoragePtr s3_storage = std::make_unique<S3ObjectStorage>(
+        auto s3_storage = std::make_unique<S3ObjectStorage>(
             std::move(cache), getClient(config, config_prefix, context),
             getSettings(config, config_prefix, context),
             uri.version_id, s3_capabilities, uri.bucket);
 
+        if (!config.getBool(config_prefix + ".skip_access_check", false))
+        {
+            if (s3_capabilities.support_batch_delete && checkBatchRemoveIsMissing(*s3_storage, uri.key))
+            {
+                s3_storage->setCapabilitiesSupportBatchDelete(false);
```

Comment:
Deserves a comment

Reply:
Sorry, I meant log message

Reply:
@excitoon you don't wan't see a message in this case? Can you elaborate why?

Reply:
Yes I want. I just missed that tiny resolved conversation.